### PR TITLE
Add the `column` filter

### PR DESCRIPTION
**Pull request by willrowe**
This adds a `column` filter that uses the [`array_column`](http://us2.php.net/manual/en/function.array-column.php) function added in PHP 5.5. Test included.

### CODE REVIEW

File: lib/Twig/Extension/Core.php [hunk 1]
```diff
@@ -617,6 +618,33 @@ function twig_array_merge($arr1, $arr2)
...
+ *
+ * @param array|Traversable $arr        An array
+ * @param mixed             $column_key The column key
+ *
+ * @return array The array of values
+ */
+function twig_array_column($arr, $column_key)
+{
+    if ($arr instanceof Traversable) {
+        $arr = iterator_to_array($arr1);
```
Comment (raziel057) [inline lib/Twig/Extension/Core.php]: `$arr1` is undefined here. It must be `$arr`.

Comment (willrowe) [reply lib/Twig/Extension/Core.php]: Good catch, I'm updating it with the fix and a test.

**Comment by fabpot** [comment]
This PR was auto-closed because I've removed the master branch and this PR was based on master. I cannot reopen it unfortunately.

**Comment by fabpot** [comment]
Changing the base branch to 2.x worked :)

**Comment by PGLongo** [comment]
@fabpot there is a reason why this PR hasn't been merged? I ask because I was doing the same PR

File: lib/Twig/Extension/Core.php [hunk 2]
```diff
@@ -611,6 +612,33 @@ function twig_array_merge($arr1, $arr2)
...
+ *
+ * @param array|Traversable $arr        An array
+ * @param mixed             $column_key The column key
+ *
+ * @return array The array of values
+ */
+function twig_array_column($arr, $column_key)
+{
+    if ($arr instanceof Traversable) {
+        $arr = iterator_to_array($arr);
```
Comment (xabbuh) [inline lib/Twig/Extension/Core.php]: This will consume the iterator if it is not rewindable, right? Is that something the user will actually expect?

Comment (jvasseur) [reply lib/Twig/Extension/Core.php]: Sine the purpose of the filter is to get some data from the iterator, I think consuming it is the expected behavior.

Comment (willrowe) [reply lib/Twig/Extension/Core.php]: I added this to bring the feature in line with the [random](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L299), [replace](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L450), [merge](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L548-L554), [slice](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L582-L588), [join](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L651), [reverse](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L779), [sort](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L815), and [batch](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L1417) functions/filters.

Comment (PGLongo) [reply lib/Twig/Extension/Core.php]: @xabbuh the behaviour is the same of [random](https://github.com/twigphp/Twig/blob/60a8f465d6f259c92a8fe0fb61cf8f767bd8d025/lib/Twig/Extension/Core.php#L299)

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
@willrowe Can you rebase on current 2.x and add a note in the CHANGELOG file before I merge? Thank you.

**Comment by willrowe** [comment]
Sure, I can do that at some point this week. Thanks for taking a look at this!

**Comment by willrowe** [comment]
@fabpot rebased and ready to go.

**Comment by fabpot** [comment]
Thank you @willrowe.

**Pull request merged**